attestation-agent/attestation-agent/src/bin/grpc-aa/server.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::*;
+use attestation::attestation_agent_service_server::{
+    AttestationAgentService, AttestationAgentServiceServer,
+};
+use attestation::{
+    ExtendRuntimeMeasurementRequest, ExtendRuntimeMeasurementResponse, GetEvidenceRequest,
+    GetEvidenceResponse, GetTokenRequest, GetTokenResponse,
+};
+use attestation_agent::{AttestationAPIs, AttestationAgent};
+use log::{debug, error};
+use std::net::SocketAddr;
+use tokio::sync::Mutex;
+use tonic::{transport::Server, Request, Response, Status};
+
+use self::attestation::{CheckInitDataRequest, CheckInitDataResponse};
+
+mod attestation {
+    tonic::include_proto!("attestation_agent");
+}
+
+pub const AGENT_NAME: &str = "attestation-agent";
+
+#[derive(Default)]
+pub struct AA {
+    inner: Mutex<AttestationAgent>,
+}
+
+#[tonic::async_trait]
+impl AttestationAgentService for AA {
+    async fn get_token(
+        &self,
+        request: Request<GetTokenRequest>,
+    ) -> Result<Response<GetTokenResponse>, Status> {
+        let request = request.into_inner();
+
+        let mut attestation_agent = self.inner.lock().await;
+
+        debug!("Call AA to get token ...");
+
+        let token = attestation_agent
+            .get_token(&request.token_type)
+            .await
+            .map_err(|e| {
+                error!("Call AA to get token failed: {}", e);
+                Status::internal(format!("[ERROR:{}] AA get token failed: {}", AGENT_NAME, e))
```

Comment:
there is a general question about whether it's a good idea to pass "rich" error messages to the consumer on API endpoints. in the AA case I'm not sure, since this is supposed to be "local" service. In any case it would probably be propagated not only to the kata-agent log, but also to a api-server-rest client.

Reply:
Yes, right. My personal idea is to handle all the error information issues in a separate PR. This error information is just copied from the old code.

wdyt?

Reply:
yes, agreed to look at this globally.